### PR TITLE
Code Quality: Cleaned up Files.Backend

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Motivation & Context

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #12340
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

File: src/Files.Backend/Data/Enums/AddItemDialogItemType.cs [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 {
 	public enum AddItemDialogItemType
 	{
 		Folder = 0,
+
 		File = 1,
+
 		Shortcut = 2,
+
 		Cancel = 3
```
Comment (yaira2) [inline src/Files.Backend/Data/Enums/AddItemDialogItemType.cs:14]: I'm not that into adding lines between enums

File: src/Files.Backend/Enums/AppEnvironment.cs [hunk 2]
```diff
@@ -1,11 +0,0 @@
-﻿
-namespace Files.Backend.Enums
-{
-	public enum AppEnvironment
```
Comment (yaira2) [inline src/Files.Backend/Enums/AppEnvironment.cs:4]: Where was this moved to instead?

File: src/Files.Backend/Data/Enums/AddItemDialogItemType.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/AddItemDialogItemType.cs:14]: This is phase 1. Preparing for adding xml comments.

Comment (yaira2) [reply src/Files.Backend/Data/Enums/AddItemDialogItemType.cs:14]: If the next phase is just comments, feel free to push another commit.

Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/AddItemDialogItemType.cs:14]: But I have decided what I do in which PR.

> 1. Rename files with the appropriate name and put them into appropriate namespace for each.
> 2. Rename variables with preferred name to follow the guideline (The most likely to happen is that I rename property-backing-field from '_isHidden' to '_IsHidden')
> 3. Reorder variables & methods to follow the class structure guideline (Section 3 in the guildeline)
> 4. Add/Update/Improve C# XML Comments and double-slashed comments.
> 5. Refactor to improve performance. You can see what I will work on in Requirements section.

Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/AddItemDialogItemType.cs]: Done. (I also did some refactoring/reordering)

File: src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs [hunk 3]
```diff
@@ -3,12 +3,24 @@
...
+	/// </summary>
 	public enum SearchBoxTextChangeReason : ushort
 	{
+		/// <summary>
+		/// The user has changed text.
+		/// </summary>
 		UserInput,
 
+		/// <summary>
+		/// The Files has programatically changed.
```
Comment (yaira2) [inline src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: *files

File: src/Files.Backend/Data/Enums/AddItemDialogItemType.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.Backend/Data/Enums/AddItemDialogItemType.cs]: I'm ready for the next commit

File: src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs (hunk 3 above)
Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: I meant the Files app. Should I change to The app has…?

Comment (yaira2) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: What does it mean by "changed"

Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: changed text in SearchBox.

Comment (yaira2) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: How about "Search box text has changed"?

Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]: Yeah, there’s some reasons why it is changed. UserInput (the user has changed it by their keyboard or sth, Programmatic (The app has changed it, like autofill or sth), SuggestChosen(the user has chosen a suggest of AutoSuggestTextBox.

File: src/Files.Backend/CommandLine/CommandLineParser.cs [hunk 4]
```diff
@@ -1,29 +1,44 @@
...
+
 			return ParseSplitArguments(parsedArgs);
 		}
 
+		/// <summary>
+		/// Parses raw command line string.
+		/// </summary>
+		/// <param name="cmdLineStrings">String of command line to parse.</param>
+		/// <returns>A collection of parsed command.</returns>
 		public static ParsedCommands ParseUntrustedCommands(string[] cmdLineStrings)
```
Comment (ferrariofilippo) [inline src/Files.Backend/CommandLine/CommandLineParser.cs:30]: `param` is not a `String`

File: src/Files.Backend/CommandLine/CommandLineParser.cs [hunk 5]
```diff
@@ -1,29 +1,44 @@
...
+
 			return ParseSplitArguments(parsedArgs);
 		}
 
+		/// <summary>
+		/// Convert string split arguments into a collection of <see cref="ParsedCommand"/>
+		/// </summary>
+		/// <param name="parsedArgs">String of command line to parse.</param>
+		/// <returns>A collection of parsed command.</returns>
 		private static ParsedCommands ParseSplitArguments(List<KeyValuePair<string, string[]>> parsedArgs)
```
Comment (ferrariofilippo) [inline src/Files.Backend/CommandLine/CommandLineParser.cs:42]: As above

File: src/Files.Backend/Data/Enums/ImpossibleActionResponseTypes.cs [hunk 6]
```diff
@@ -1,8 +1,11 @@
 ﻿// Copyright (c) 2023 Files Community
 // Licensed under the MIT License. See the LICENSE.
 
-namespace Files.Backend.Enums
+namespace Files.Backend.Data.Enums
 {
+	/// <summary>
+	/// Defines constants that specify 
+	/// </summary>
```
Comment (ferrariofilippo) [inline src/Files.Backend/Data/Enums/ImpossibleActionResponseTypes.cs:8]: I think you forgot this

File: src/Files.Backend/Data/Enums/PreviewPaneStates.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+		/// <summary>
+		/// No preview available status.
+		/// </summary>
+		NoPreviewAvailable,
+
+		/// <summary>
+		/// No preview or details status.
+		/// </summary>
+		NoPreviewOrDetailsAvailable,
```
Comment (ferrariofilippo) [inline src/Files.Backend/Data/Enums/PreviewPaneStates.cs:24]:
```suggestion
		/// <summary>
		/// No preview or details available status.
		/// </summary>
		NoPreviewOrDetailsAvailable,
```

File: src/Files.Backend/ViewModels/Dialogs/AddItemDialog/AddItemDialogViewModel.cs [hunk 8]
```diff
@@ -1,65 +1,63 @@
...
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Threading.Tasks;
 
 namespace Files.Backend.ViewModels.Dialogs.AddItemDialog
 {
 	public sealed class AddItemDialogViewModel : ObservableObject
 	{
-		public IImageService ImagingService { get; } = Ioc.Default.GetRequiredService<IImageService>();
+		private readonly IImageService _imagingService = Ioc.Default.GetRequiredService<IImageService>();
```
Comment (ferrariofilippo) [inline src/Files.Backend/ViewModels/Dialogs/AddItemDialog/AddItemDialogViewModel.cs:10]: Shouldn't you move `Ioc.Default.GetRequiredService<IImageService>()` to the constructor?

File: src/Files.Backend/Helpers/FileExtensionHelpers.cs [hunk 9]
```diff
@@ -1,156 +1,176 @@
...
 			return extensions.Any(ext => filePathToCheck.EndsWith(ext, StringComparison.OrdinalIgnoreCase));
 		}
 
 		/// <summary>
 		/// Check if the file extension is an image file.
 		/// </summary>
 		/// <param name="fileExtensionToCheck">The file extension to check.</param>
-		/// <returns><c>true</c> if the fileExtensionToCheck is an image;
-		/// otherwise, <c>false</c>.</returns>
+		/// <returns><c>true</c> if the fileExtensionToCheck is an image; otherwis,e, <c>false</c>.</returns>
```
Comment (ferrariofilippo) [inline src/Files.Backend/Helpers/FileExtensionHelpers.cs:29]: As above

File: src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs [hunk 10]
```diff
@@ -82,25 +84,40 @@ public struct WIN32_FIND_DATA
...
 			FINDEX_INFO_LEVELS findInfoLevel = FINDEX_INFO_LEVELS.FindExInfoBasic;
+
 			int additionalFlags = FIND_FIRST_EX_LARGE_FETCH;
 
-			IntPtr hFile = FindFirstFileExFromApp(targetPath, findInfoLevel, out findData, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, additionalFlags);
+			IntPtr hFile = FindFirstFileExFromApp(
+				targetPath,
+				findInfoLevel,
+				out findData, FINDEX_SEARCH_OPS.FindExSearchNameMatch,
+				IntPtr.Zero, additionalFlags);
```
Comment (ferrariofilippo) [inline src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs:112]: I think you should split these two lines into four

File: src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs [hunk 11]
```diff
@@ -1,12 +1,14 @@
...
-using System;
 using System.Runtime.InteropServices;
 using System.Runtime.InteropServices.ComTypes;
 
 namespace Files.Backend.Helpers
 {
+	/// <summary>
+	/// NativeFindStorageItemHelper
+	/// </summary>
 	public class NativeFindStorageItemHelper
```
Comment (ferrariofilippo) [inline src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs:12]: I think you should add something, written like this it provides no information

File: src/Files.Backend/CommandLine/CommandLineParser.cs (hunk 4 above)
Comment (0x5bfa) [reply src/Files.Backend/CommandLine/CommandLineParser.cs:30]: Fixed

File: src/Files.Backend/CommandLine/CommandLineParser.cs (hunk 5 above)
Comment (0x5bfa) [reply src/Files.Backend/CommandLine/CommandLineParser.cs:42]: Fixed

File: src/Files.Backend/Data/Enums/PreviewPaneStates.cs (hunk 7 above)
Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/PreviewPaneStates.cs:24]: Fixed

File: src/Files.Backend/Data/Models/VolumeInfo.cs [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+		public string Id
+			=> $@"\\?\Volume{{{Guid}}}";
+
+		public Guid Guid { get; }
+
+		public VolumeInfo(Guid guid)
+			=> Guid = guid;
+
+		public VolumeInfo(string id)
+			=> Guid = ToGuid(id);
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.Backend/Data/Models/VolumeInfo.cs:25]: Fixed

File: src/Files.Backend/Data/Models/VolumeInfo.cs [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+			=> Id;
+
+		public override int GetHashCode()
+			=> Guid.GetHashCode();
+
+		public override bool Equals(object? other)
+			=> other is VolumeInfo info && Equals(info);
+
+		public bool Equals(VolumeInfo other)
+			=> other.Guid.Equals(Guid);
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.Backend/Data/Models/VolumeInfo.cs:49]: Fixed

File: src/Files.Backend/ViewModels/Dialogs/AddItemDialog/AddItemDialogViewModel.cs (hunk 8 above)
Comment (0x5bfa) [reply src/Files.Backend/ViewModels/Dialogs/AddItemDialog/AddItemDialogViewModel.cs:10]: Fixed

File: src/Files.Backend/Helpers/FileExtensionHelpers.cs [hunk 14]
```diff
@@ -1,156 +1,176 @@
...
 	public static class FileExtensionHelpers
 	{
 		/// <summary>
 		/// Check if the file extension matches one of the specified extensions.
 		/// </summary>
 		/// <param name="filePathToCheck">Path or name or extension of the file to check.</param>
 		/// <param name="extensions">List of the extensions to check.</param>
-		/// <returns><c>true</c> if the filePathToCheck has one of the specified extensions;
-		/// otherwise, <c>false</c>.</returns>
+		/// <returns><c>true</c> if the filePathToCheck has one of the specified extensions; otherwis,e, <c>false</c>.</returns>
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.Backend/Helpers/FileExtensionHelpers.cs:16]: Fixed

File: src/Files.Backend/Helpers/FileExtensionHelpers.cs (hunk 9 above)
Comment (0x5bfa) [reply src/Files.Backend/Helpers/FileExtensionHelpers.cs:29]: Fixed

File: src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs (hunk 10 above)
Comment (0x5bfa) [reply src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs:112]: Fixed

File: src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs (hunk 11 above)
Comment (0x5bfa) [reply src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs:12]: Fixed

Comment (0x5bfa) [reply src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs:12]: @lukeblevins I think we should remove this class and introduce Vanara in this project. What do you think?

Following files can be replaced with Vanara like you introduced in your latest PR into Files.App.Storeage.

- NativeFindStorageItemHelper.cs
- NativeFileOperationsHelper.cs
- NativeWinApiHelper.cs
- NativeIoDeviceControlHelper.cs
- NativeDirectoryChangesHelper.cs
- InteropHelpers.cs
- FileUtils.cs

**Review by ferrariofilippo — APPROVED** [review]
LGTM

File: src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs (hunk 11 above)
Comment (0x5bfa) [reply src/Files.Backend/Helpers/NativeFindStorageItemHelper.cs]: @ferrariofilippo What would you think?

File: src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+	/// </summary>
+	public enum SearchBoxTextChangeReason : ushort
+	{
+		/// <summary>
+		/// The user has changed text manually.
+		/// </summary>
+		UserInput,
+
+		/// <summary>
+		/// The app has changed programmatically.
```
Comment (yaira2) [inline src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]:
```suggestion
		/// The search text has changed programmatically.
```

Comment (0x5bfa) [reply src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs]: Fixed

**Comment by 0x5bfa** [comment]
@yaira2 Can you review again?

File: src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Files.Backend.Data.Enums
+{
+	/// <summary>
+	/// Defines constants that specify search box text changed reason.
+	/// </summary>
+	public enum SearchBoxTextChangeReason : ushort
+	{
+		/// <summary>
+		/// The SearchBox text has been changed manually.
```
Comment (yaira2) [inline src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:12]:
```suggestion
		/// The SearchBox text was manually changed.
```

File: src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+		/// </summary>
+		UserInput,
+
+		/// <summary>
+		/// The SearchBox text has been changed programmatically.
```
Comment (yaira2) [inline src/Files.Backend/Data/Enums/SearchBoxTextChangeReason.cs:17]:
```suggestion
		/// The SearchBox text was changed programmatically.
```

**Review by yaira2 — APPROVED** [review]

**Comment by yaira2** [comment]
Ready for the next commit

**Comment by 0x5bfa** [comment]
I wanna split up. So that was last commit I suppose.

**Pull request merged**